code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -150,21 +150,8 @@
 /obj/machinery/atmospherics/components/unary/thermomachine/default_change_direction_wrench(mob/user, obj/item/I)
 	if(!..())
 		return FALSE
-	SetInitDirections()
-	var/obj/machinery/atmospherics/node = nodes[1]
-	if(node)
-		if(src in node.nodes) //Only if it's actually connected. On-pipe version would is one-sided.
-			node.disconnect(src)
-		nodes[1] = null
-	if(parents[1])
-		nullifyPipenet(parents[1])
-
-	atmosinit()
-	node = nodes[1]
-	if(node)
-		node.atmosinit()
-		node.addMember(src)
-	SSair.add_to_rebuild_queue(src)
+	set_init_directions()
+	update_icon()
```

Comment:
Should this be calling ``update_appearance(UPDATE_ICON)`` instead, or is there a specific reason it's gotta call this?